src/structures/interfaces/Collector.js

```diff
@@ -147,6 +147,38 @@ class Collector extends EventEmitter {
     });
   }
 
+  /**
+   * Allows this collector to be consumed with `for await...of`
+   * @type {AsyncIterator}
+   */
+  async * [Symbol.asyncIterator]() {
+    const queue = [];
```

Comment:
using an unbounded queue will cause a memory leak if there’s a bottleneck or a hanging promise downstream, maybe include a check to see if the queue is larger than a specific number

Reply:
No, that would be adding opinion to Collectors. The user can already set a max number to collect on their own.